pkg/certmaker/certmaker.go

```diff
@@ -0,0 +1,464 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package certmaker implements a certificate creation utility for Fulcio.
+// It supports creating root, intermediate, and leaf certs using (AWS, GCP, Azure, HashiVault).
+package certmaker
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature/kms"
+	"github.com/sigstore/sigstore/pkg/signature/options"
+
+	// Initialize AWS KMS provider
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/aws"
+	// Initialize Azure KMS provider
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/azure"
+	// Initialize GCP KMS provider
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/gcp"
+	// Initialize HashiVault KMS provider
+	_ "github.com/sigstore/sigstore/pkg/signature/kms/hashivault"
+	"go.step.sm/crypto/x509util"
+)
+
+type signerWrapper struct {
```

Comment:
Is this needed? the KMS implementations implement Public and Sign - https://github.com/sigstore/sigstore/blob/main/pkg/signature/kms/gcp/signer.go#L144-L164

Reply:
ah, good point / will remove and extend.